File: event-service/src/main/java/com/greenlight/event/service/ReadOnlyService.java
package com.greenlight.event.service;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

/**
 * Base interface for services
 *
 * @param <T> The base Type of repository
 * @param <I> Id Type
 * @author gvart
 */
public interface ReadOnlyService<T,I> {
  Flux<T> findAll();

  Mono<T> findOne(I id);

}

File: event-service/src/main/kotlin/com/greenlight/event/service/ReadOnlyService.kt
package com.greenlight.event.service

import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

/**
 * Base interface for services
 *
 * @param <T> The base Type of repository
 * @param <I> Id Type
 * @author gvart
 */
interface ReadOnlyService<T, I> {
    fun findAll(): Flux<T>

    fun findOne(id: I): Mono<T>

}
